libs/ledger-live-common/src/exchange/providers/swap.ts

```diff
@@ -145,22 +144,54 @@ export const getProvidersData = async () => {
         "https://crypto-assets-service.api.aws.prd.ldg-tech.com/v1/partners?output=name,payload_signature_computed_format,signature,public_key,public_key_curve",
       )
     ).json();
-    return transformData(providersData);
+    return providersData;
   } catch {
     return swapProviders;
   }
 };
 
-export const getProvidersAdditionalData = (providerName: string): AdditionalProviderConfig => {
-  const res = swapAdditionData[providerName.toLowerCase()];
-
-  if (!res) {
-    throw new Error(`Unknown partner ${providerName}`);
+export const getProvidersCDNData = async () => {
+  try {
+    const providersData = await (
+      await fetch("https://cdn.live.ledger.com/swap-providers/data.json")
+    ).json();
+    return providersData;
+  } catch {
+    return swapAdditionData;
   }
+};
+
+export const fetchAndMergeProviderData = async () => {
```

Comment:
You maybe should consider use libs like TanStackQuery to achieve this kind of stuffs :)

I mean Fetch + Merge data